pkg/segment/tracing/handler/tracehandler.go

```diff
@@ -0,0 +1,344 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"regexp"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	pipesearch "github.com/siglens/siglens/pkg/ast/pipesearch"
+	"github.com/siglens/siglens/pkg/es/writer"
+	"github.com/siglens/siglens/pkg/segment/tracing/structs"
+	"github.com/siglens/siglens/pkg/segment/tracing/utils"
+	putils "github.com/siglens/siglens/pkg/utils"
+	log "github.com/sirupsen/logrus"
+	"github.com/valyala/fasthttp"
+)
+
+func ProcessSearchTracesRequest(ctx *fasthttp.RequestCtx, myid uint64) {
+
+	rawJSON := ctx.PostBody()
+	if rawJSON == nil {
+		log.Errorf(" ProcessPipeSearchRequest: received empty search request body ")
+		pipesearch.SetBadMsg(ctx)
+		return
+	}
+
+	readJSON := make(map[string]interface{})
+	var jsonc = jsoniter.ConfigCompatibleWithStandardLibrary
+	decoder := jsonc.NewDecoder(bytes.NewReader(rawJSON))
+	decoder.UseNumber()
+	err := decoder.Decode(&readJSON)
+	if err != nil {
+		ctx.SetStatusCode(fasthttp.StatusBadRequest)
+		_, err = ctx.WriteString(err.Error())
+		if err != nil {
+			log.Errorf("ProcessPipeSearchRequest: could not write error message err=%v", err)
+		}
+		log.Errorf("ProcessPipeSearchRequest: failed to decode search request body! Err=%v", err)
+	}
+
+	nowTs := putils.GetCurrentTimeInMs()
+	searchText, startEpoch, endEpoch, _, _, _ := pipesearch.ParseSearchBody(readJSON, nowTs)
+	if err != nil {
+		log.Errorf("ProcessSearchTracesRequest: failed to parse search body  err=%v", err)
+
+		_, wErr := ctx.WriteString(err.Error())
+		if wErr != nil {
+			log.Errorf("ProcessSearchTracesRequest: could not write error message! %v", wErr)
+		}
+		return
+	}
+
+	// Parse the JSON data from ctx.PostBody
+	var requestData map[string]interface{}
+	if err := json.Unmarshal(ctx.PostBody(), &requestData); err != nil {
+		log.Errorf("ProcessSearchTracesRequest: could not unmarshal json body, err=%v", err)
+		return
+	}
```

Comment:
I think you can make this both much more compact and more readable by defining a struct that has the fields we expect to get from this `RequestCtx` and unmarshaling the `ctx.PostBody()` into that struct, so you should do that. This will also help avoid potential typos because later when you access this `requestData` you would do something like `requestData.SearchText` instead of `requestData["searchText"]`

Reply:
Since this method is belong to the PR of search traces. I will modify that in that PR.